src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const tempState = {};
+  const copyState = [];
+
+  Object.assign(tempState, state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(tempState, action.extraData);
+
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete tempState[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        for (const key of Object.keys(tempState)) {
+          delete tempState[key];
+        }
+        break;
+      }
+
+      default: {
+        return 'Unexpected type of action';
```

Comment:
Returning a string 'Unexpected type of action' in the default case of the switch statement is not ideal for error handling. It would be better to throw an error instead, as it would immediately notify the developer of the issue rather than silently failing.